app/models/mixins/supports_feature_mixin.rb

```diff
@@ -116,6 +116,30 @@ def supports?(feature)
       public_send("supports_#{feature}?")
     end
 
+    # all subclasses that are considered for supporting features
+    def supported_subclasses
+      descendants
+    end
+
+    def subclasses_supporting(feature)
+      supported_subclasses.select { |subclass| subclass.supports?(feature) }
+    end
+
+    # scope to query all those classes that support a particular feature
+    def supporting(feature)
+      where(:type => types_supporting(feature))
+    end
+
+    def types_supporting(feature)
+      subclasses_supporting(feature).map(&:name)
+    end
+
+    # Ems types (class names) that support this feature
+    # Ems.where(:type => Host.ems_types_supporting(:create))
+    def ems_types_supporting(feature)
+      subclasses_supporting(feature).map(&:module_parent).map(&:name)
```

Comment:
This is perfect for querying on `.where(:type =>)` but we might sometimes want a list of the class names, could you split this up into one that returns the class names then this becomes `emses_supporting(feature).map(&:name)`

(plural emses doesn't look great but :shrug: )

Reply:
should we name it `providers_supporting` ?

Reply:
@agrare This PR now has the `def supporting(feature)` method which returns a scope - is that what you were looking for ?

Reply:
Well that returns instances of that class where this one is about returning the EMS classes.

Reply:
Ohhh I see....yeah then I agree with you.

Reply:
And actually a scope that returns the EMS instances (similar to `.supporting(feature)`) would make these: https://github.com/ManageIQ/manageiq-ui-classic/pull/7958/files#diff-b81586b5ee3afef460d25dce2deda502148e37faba61591f771df88c23128daeR32 even simpler

Reply:
`ManageIQ::Providers::CloudManager::AuthKeyPair.providers_supporting(:create)` for example

Reply:
I added the providers_supporting that returns classes, but to be honest, I do not see us needing types_supporting as I think `supporting()` will meet our needs.

I also do not think we need `providers_supporting`. Wonder if we want to convert `provider_types_supporting` into a scope and we could also do a scope instead. merge would add to another chain.

I guess having strings is good because we store stuff in the session

Reply:
Yeah I like the scope over something that returns the class name strings just to then pass that to a `.where(:type =>`

Reply:
@agrare drop `type_supporting`?

Thinking `provider_type_supporting` may stay for a little until we get a good grasp of it.

Reply:
ok, added `providers_supporting`
This removes our need for looking at provider types.
if we need part of this we can scope merge it